direct/tpm2/tpm2.go

```diff
@@ -387,6 +387,41 @@ type CreateLoadedResponse struct {
 // Response implements the Response interface.
 func (*CreateLoadedResponse) Response() tpm.CC { return tpm.CCCreateLoaded }
 
+// Hash is the input to TPM2_Hash.
+// See definition in Part 3, Commands, section 15.4
+type Hash struct {
+	//data to be hashed
+	Data tpm2b.MaxBuffer
+	// algorithm for the hash being computed - shall not be TPM_ALH_NULL
+	HashAlg tpmi.AlgHash
+	// hierarchy to use for the tickey (TPM_RH_NULL_allowed)
+	Hierarchy tpmi.RHHierarchy
```

Comment:
If this member is omitted, will TPM_RH_NULL be used by default?

Reply:
Regardless, we should have a test that using a Null Hierarchy with `TPM2_Hash` and `TPM2_SequenceComplete` works as expected.

Reply:
> If this member is omitted, will TPM_RH_NULL be used by default?

Yes, it has now been added. The tests are being made now.